kumascript/macros/Compat.ejs

```diff
@@ -30,6 +30,11 @@ Example calls
 
 */
 
+if ($0) {
+  throw new Error(`Calling the Compat macro with any arguments is
+  deprecated; instead use the 'browser-compat' front-matter key.`);
```

Comment:
Could we reuse `mdn.deprecated()` from `kumascript/src/api/mdn.js` by adding an optional `message` parameter?

```
  deprecated(message = "This macro has been deprecated, and should be removed.") {
    this.env.recordNonFatalError(
      "deprecated",
      message
    );
  },
```

Reply:
@sideshowbarker If you merge the latest `main`, you will be able to pass a message to `mdn.deprecated()`.